test/integration/integration_test.go

```diff
@@ -303,6 +310,74 @@ func (i *integrationTest) exerciseIncrementalBuild(tag, imageName string, remove
 	}
 }
 
+func TestOnBuildTrigger(t *testing.T) {
+	integration(t).exerciseOnBuildTrigger(TagOnBuildTrigger, false, FakeImageOnBuild, FakeScriptsFileURL)
+}
+
+func (i *integrationTest) exerciseOnBuildTrigger(tag string, verifyCallback bool, imageName string, scriptsURL string) {
+	t := i.t
+	callbackURL := ""
+	callbackInvoked := false
+	callbackHasValidJSON := false
+	if verifyCallback {
+		handler := func(w http.ResponseWriter, r *http.Request) {
+			// we got called
+			callbackInvoked = true
+			// the header is as expected
+			contentType := r.Header["Content-Type"][0]
+			callbackHasValidJSON = contentType == "application/json"
+			// the request body is as expected
+			if callbackHasValidJSON {
+				defer r.Body.Close()
+				body, _ := ioutil.ReadAll(r.Body)
+				type CallbackMessage struct {
+					Payload string
+					Success bool
+				}
+				var callbackMessage CallbackMessage
+				err := json.Unmarshal(body, &callbackMessage)
+				callbackHasValidJSON = (err == nil) && (callbackMessage.Success)
+			}
+		}
+		ts := httptest.NewServer(http.HandlerFunc(handler))
+		defer ts.Close()
+		callbackURL = ts.URL
+	}
+
+	config := &api.Config{
+		DockerConfig: dockerConfig(),
+		BuilderImage: imageName,
+		Source:       TestSource,
+		Tag:          tag,
+		Incremental:  false,
+		CallbackURL:  callbackURL,
+		ScriptsURL:   scriptsURL}
+
+	b, err := strategies.GetStrategy(config)
+	if err != nil {
+		t.Fatalf("Cannot create a new builder.")
+	}
+	resp, err := b.Build(config)
+	if err != nil {
+		t.Fatalf("An error occurred during the build: %v", err)
+	} else if !resp.Success {
+		t.Fatalf("The build failed.")
+	}
+	if callbackInvoked != verifyCallback {
+		t.Fatalf("Sti build did not invoke callback")
+	}
+	if callbackHasValidJSON != verifyCallback {
+		t.Fatalf("Sti build did not invoke callback with valid json message")
+	}
+
+	i.checkForImage(tag)
+	containerID := i.createContainer(tag)
+	defer i.removeContainer(containerID)
+	i.checkBasicBuildState(containerID, resp.WorkingDir)
+
+	i.fileExists(containerID, "/sti-fake/src/onbuild")
```

Comment:
rather than create a whole new exercise* function just for this(? did i miss some other delta?), i would augment the existing exercise* functions to enable them to validate the output image by checking for file presence.

You shouldn't really need special test code to test an ONBUILD image, you should just be able to confirm that the existing tests pass when the input image contains ONBUILD instructions (and confirm the ONBUILD instructions were executed)